soh/src/code/z_play.c

```diff
@@ -242,17 +242,19 @@ void GivePlayerRandoRewardZeldaLightArrowsGift(GlobalContext* globalCtx, Randomi
         (gEntranceTable[((void)0, gSaveContext.entranceIndex)].scene == SCENE_TOKINOMA) &&
         !Flags_GetTreasure(globalCtx, 0x1E) && player != NULL && !Player_InBlockingCsMode(globalCtx, player) &&
         globalCtx->sceneLoadFlag == 0) {
-        GetItemID getItemId = Randomizer_GetItemIdFromKnownCheck(check, GI_ARROW_LIGHT);
-        GiveItemWithoutActor(globalCtx, getItemId);
-        player->pendingFlag.flagID = 0x1E;
-        player->pendingFlag.flagType = FLAG_SCENE_TREASURE;
+        GetItemEntry getItem = Randomizer_GetItemFromKnownCheck(check, GI_ARROW_LIGHT);
+        if (player->pendingFlag.flagType == FLAG_NONE && GiveItemWithoutActor(globalCtx, getItem.getItemId)) {
```

Comment:
Do we need to do this flag check and set via pending flags for the other GivePlayerRandoReward___?

Reply:
Potentially, this was done specifically for this check as part of a fix for that specific check in the new system because it was also potentially competing with another check in the same area. The pending flags system in general was added in a previous PR.

Reply:
What I probably want to do eventually is wrap that pendingFlag check and set in a function, but that's a bit out of scope for this PR I think.